chips/atmega328p-hal/src/wdt.rs

```diff
@@ -0,0 +1,108 @@
+//! Provides a system reset when a counter reaches a given time-out value.
+//!
+//! # Example
+//! ```
+//! let mut watchdog = Wdt::new(&dp.CPU.mcusr, dp.WDT);
+//! watchdog.disable();
+//! watchdog.start(WatchdogTimeOutPeriod::Ms8000);
+//!
+//! loop {
+//!     watchdog.feed();
+//! }
+//! ```
+
+use avr_device::generic::Reg;
+use avr_hal::hal::watchdog::*;
+
+pub enum WatchdogTimeOutPeriod {
```

Comment:
`struct`s and `enum`s should eagerly derive common traits ([C-COMMON-TRAITS](https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits)).